packages/api/endpoints/collections.js

```diff
@@ -162,29 +169,38 @@ async function post(req, res) {
 async function put(req, res) {
   const { name, version } = req.params;
   const collection = req.body;
+  let dynamoRecord;
+  let oldCollection;
 
   if (name !== collection.name || version !== collection.version) {
     return res.boom.badRequest('Expected collection name and version to be'
       + ` '${name}' and '${version}', respectively, but found '${collection.name}'`
       + ` and '${collection.version}' in payload`);
   }
-
   const collectionsModel = new models.Collection();
-
-  if (!(await collectionsModel.exists(name, version))) {
+  try {
+    oldCollection = await collectionsModel.get({ name, version });
+  } catch (error) {
+    if (error.name !== 'RecordDoesNotExist') {
```

Comment:
What is the net effect of this change? If a non `RecordDoesNotExist` error is thrown, then the default endpoint failure response is used (400 I think?) instead of a 404?

Reply:
Hmm, this got removed in the merge somehow.   It should definitely `res.boom.notFound`.

Reply:
Sorted: 1541db0dd

The net sum of this change is that we're don't a `get` instead of `exists` on the documentDB client, and catching the error thrown by get explicitly to throw via boom.notFound. 

It's been a minute since I've looked at this, so I'll double back and check that we have tests around this behavior/it's catching the correct error.